CDHSH-75 Load data form the redux store into the frontend view

Liannus opened the pull request:
Changelog:
- Loaded data from the store into the view
- Changed brand logo in navbar to use material-ui hovering/clicking

Other implementation details:
- Added selectors for `userId` and `profilebyId`
- Mapped data from the store onto the props and loaded inside the layout
- Added function to take the first letter of names (for the avatar)
- Fixed routing from landing page going to /app
- Added tests for new functions inside of profile model

DevinSit commented on services/frontend/src/utils/models/Profile.test.js:
```diff
@@ -0,0 +1,46 @@
+        new Profile({id: "3", phone: "9051142315", userId: "3"})
+    ];
+
+    const userIds = [
```
What's even the point of this variable? It's not used anywhere; not to generate the test data, and not even in the tests.

Liannus replied:
I was passing in userIds[0], etc, etc. 

Turns out I changed that and don't need it anymore, so it's removed now.

Liannus commented on services/frontend/src/scenes/Profile/Profile.js:
```diff
@@ -37,10 +38,22 @@ const dummyData = [
-        projects={projects}
-    />
+//Split at each word, take the first letter and then convert them to uppercase.
+const addAvatarAcronym = (name) => (
```
Yeah, that was my intention initially. Probably do some editing to make it so names like WWW would still work.

That said, I swapped it to first and last since fixing CSS text resizing is a feature that would take a lot longer.

Liannus commented on services/frontend/src/scenes/Profile/Profile.js:
```diff
@@ -37,10 +38,22 @@ const dummyData = [
+//Split at each word, take the first letter and then convert them to uppercase.
+const addAvatarAcronym = (name) => (
+    name.split(" ").reduce((acc, word) => {
+        return acc = acc + word[0];
```
Sure, I am still using the reduce so I changed it to this (although i tried to break it up since it's a long line).

Liannus commented on services/frontend/src/scenes/Profile/Profile.js:
```diff
@@ -37,10 +38,22 @@ const dummyData = [
-        projects={projects}
-    />
+//Split at each word, take the first letter and then convert them to uppercase.
+const addAvatarAcronym = (name) => (
```
Agreed, changed.

Liannus commented on services/frontend/src/scenes/Profile/Profile.js:
```diff
@@ -37,10 +38,25 @@ const dummyData = [
-export default Profile;
+const Profile = ({projects = dummyData, userProfile, profileLoadingState}) => {
+    if (userProfile) {
+        userProfile.nameAcronym = addAvatarAcronym(userProfile.name);
```
As mentioned in the above issues, changed:

property: avatarInitials
func: generatevatarInitials

Liannus commented on services/frontend/src/scenes/Profile/ProfileLayout.js:
```diff
@@ -62,35 +63,42 @@ const ProfileContent = ({...sectionProps}) => (
-        </div>
-    </Paper>
-);
+const PersonalDetails = ({userProfile, profileLoaded}) => {
```
Yep, this was an issue I found when working on the skills section.

Fixed.

Liannus commented on services/frontend/src/scenes/Profile/ProfileLayout.js:
```diff
@@ -62,35 +63,42 @@ const ProfileContent = ({...sectionProps}) => (
-    </Paper>
-);
+const PersonalDetails = ({userProfile, profileLoaded}) => {
+    if (userProfile || profileLoaded) {
```
Good point, changed that over as well.

I also noticed that the profileRequestSlice state is `loading` not `loaded` so I changed the variable to `profileLoading`.

Liannus commented on services/frontend/src/scenes/Profile/ProfileLayout.js:
```diff
@@ -62,35 +63,42 @@ const ProfileContent = ({...sectionProps}) => (
+        );
+    }
+    else {
+        return null;
```
Sure, swapped it to the circularProgress spinner.

Liannus commented on services/frontend/src/scenes/Profile/connect.js:
```diff
@@ -0,0 +1,13 @@
+import {profilesSlice, profilesRequestsSlice} from "store/slices";
+
+const mapStateToProps = (state) => {
+    const profileLoadState = profilesRequestsSlice.fetchAll.selectors.getState(state);
```
Yep, it fits within the line length. Swapped that over and removed the return.

Liannus commented on services/frontend/src/utils/models/User.js:
```diff
@@ -1,6 +1,15 @@
 export default class User {
+    constructor({
+        id = uuidv4(),
+        name = "",
```
Removed.

Liannus commented on services/frontend/src/store/slices/profiles.slice.js:
```diff
@@ -14,4 +16,9 @@ export const profilesSlice = createSlice({
     }
 });
 
+profilesSlice.selectors.getCurrentUserProfile = createSelector(
```
Ahh, initially I didn't completely get when to use cross slice selectors but yeah that makes senses. Not completely sure how I missed that.

Moved the selector to `crossSliceSelectors`

Liannus commented on services/frontend/src/utils/models/Profile.js:
```diff
@@ -35,4 +35,14 @@ export default class Profile {
+    //Iterate over profiles checking for the current user's id, then return that user's entire profile.
+    static filterForCurrentUser(profilesById = {}, currentUserId = "") {
+        const filteredProfile = profilesById[
+            Object.keys(profilesById).filter((profileId) => (
```
Sounds good, I added a a false check and moved the array indexing operator outside of the expression.

Liannus commented on services/frontend/src/utils/models/Profile.js:
```diff
@@ -35,4 +35,14 @@ export default class Profile {
     }
+
+    //Iterate over profiles checking for the current user's id, then return that user's entire profile.
+    static filterForCurrentUser(profilesById = {}, currentUserId = "") {
```
Alright, I'll leave this for now but remember it for a future ticket that uses the profile loading.

Liannus commented:
A few extra changes, added some scaffolding to the profile SCSS and moved the renderSectionComponent since the file makes way more sense if it's above the sections it's rendering.
